client/asset/btc/btc.go

```diff
@@ -1224,31 +1228,34 @@ func (btc *ExchangeWallet) sendWithReturn(baseTx *wire.MsgTx, addr btcutil.Addre
 		return nil, nil, fmt.Errorf("not enough funds to cover minimum fee rate. %d < %d",
 			totalIn, minFee+totalOut)
 	}
+
 	changeScript, err := txscript.PayToAddrScript(addr)
 	if err != nil {
 		return nil, nil, fmt.Errorf("error creating change script: %v", err)
 	}
 	changeIdx := len(baseTx.TxOut)
 	changeOutput := wire.NewTxOut(int64(remaining-minFee), changeScript)
-	var changeAdded bool
-	isDust := dexbtc.IsDust(changeOutput, feeRate)
+	changeAdded := !dexbtc.IsDust(changeOutput, feeRate)
 	sigCycles := 1
-	if !isDust {
-		// Add the change output with a recalculated fees
-		size += dexbtc.P2WPKHOutputSize
```

Comment:
I think this is a worthwhile change in both btc and dcrd since it's probably best not to assume a certain address format.